src/components/UI/ConfirmDialog.vue

```diff
@@ -0,0 +1,91 @@
+<script lang="ts" setup>
+import { ref } from 'vue'
+import BaseButton from '/@/components/UI/BaseButton.vue'
+
+const dialogRef = ref<HTMLDialogElement>()
+
+const open = () => {
+  dialogRef.value?.showModal()
+  dialogRef.value?.addEventListener('click', listener)
+}
+
+const close = () => {
+  dialogRef.value?.close()
+  dialogRef.value?.removeEventListener('click', listener)
+}
+
+const listener = (e: Event) => {
+  if (dialogRef.value === undefined) return
+  if (e.target === dialogRef.value) {
+    emit('close')
+  }
+}
+
+defineExpose({
+  open,
+  close
+})
+
+const emit = defineEmits<{
+  (e: 'close'): void
+  (e: 'remove'): void
+}>()
+</script>
+
+<template>
+  <dialog ref="dialogRef" :class="$style.modal">
+    <div :class="$style.container">
+      <h1 :class="$style.header">コンテストの削除</h1>
+      <p :class="$style.body">
+        コンテンストと、コンテストに含まれるチームをすべて削除します。この操作は取り消せません。
+      </p>
+      <div :class="$style.buttonContent">
+        <base-button
+          type="secondary"
+          icon="mdi:arrow-left"
+          @click="() => emit('close')"
```

Comment:
Reactじゃないのでこれでよさそうです
```suggestion
          @click="emit('close')"
```